packages/core/src/js/feedback/FeedbackFormScreen.tsx

```diff
@@ -0,0 +1,102 @@
+import { captureFeedback } from '@sentry/core';
+import type { SendFeedbackParams } from '@sentry/types';
+import * as React from 'react';
+import type { KeyboardTypeOptions } from 'react-native';
+import { Alert, Text, TextInput, TouchableOpacity, View } from 'react-native';
+
+import defaultStyles from './FeedbackFormScreen.styles';
+import type { FeedbackFormScreenProps, FeedbackFormScreenState } from './FeedbackFormScreen.types';
+
+/**
+ * Implements a feedback form screen that sends feedback to Sentry using Sentry.captureFeedback.
+ */
+export class FeedbackFormScreen extends React.Component<FeedbackFormScreenProps, FeedbackFormScreenState> {
+  public constructor(props: FeedbackFormScreenProps) {
+    super(props);
+    this.state = {
+      name: '',
+      email: '',
+      description: '',
+    };
+  }
+
+  public handleFeedbackSubmit: () => void = () => {
+    const { name, email, description } = this.state;
+    const { closeScreen, text } = this.props;
+
+    const trimmedName = name?.trim();
+    const trimmedEmail = email?.trim();
+    const trimmedDescription = description?.trim();
+
+    if (!trimmedName || !trimmedEmail || !trimmedDescription) {
+      const errorMessage = text?.formError || 'Please fill out all required fields.';
+      Alert.alert(text?.errorTitle || 'Error', errorMessage);
+      return;
+    }
+
+    if (!this._isValidEmail(trimmedEmail)) {
+      const errorMessage = text?.emailError || 'Please enter a valid email address.';
+      Alert.alert(text?.errorTitle || 'Error', errorMessage);
+      return;
+    }
+
+    const userFeedback: SendFeedbackParams = {
+      message: trimmedDescription,
+      name: trimmedName,
+      email: trimmedEmail,
+    };
+
+    captureFeedback(userFeedback);
+    closeScreen();
```

Comment:
Let's add default `onFormSubmitted` behaviour where success message is show similar to web.

<img width="272" alt="Screenshot 2024-12-05 at 18 23 46" src="https://github.com/user-attachments/assets/969f2c98-8491-4b80-a98a-dc14eaff3c66">

I think in our case it will be just centered text in place of the form.

Reply:
Since a proper toast seems to need a custom implementation (or a new library import) I added a simple alert as an initial implementation with 39a67bd9ee34ef95abcb247021e32526e01821ff
I've opened an issue https://github.com/getsentry/sentry-react-native/issues/4356 to revisit this separately.

<img width="276" alt="Screenshot 2024-12-11 at 11 30 48 AM" src="https://github.com/user-attachments/assets/993fddff-de06-41b5-9e79-f4b39fa8471c">